### DISCUSSION TITLE
undefined reference to `FT_Set_Default_Properties'

### DISCUSSION THREAD

**Discussion by Matiz7** [ImageMagick/ImageMagick#5734] [General] [answered]
It work well when I build/make on centos8
but build in docker.. 
  CXXLD    utilities/magick
MagickCore/.libs/libMagickCore-7.Q16HDRI.so: undefined reference to `FT_Set_Default_Properties'
collect2: error: ld returned 1 exit status
make[1]: *** [utilities/magick] Error 1
Makefile:8146: recipe for target 'utilities/magick' failed
Makefile:6202: recipe for target 'all' failed
make: *** [all] Error 2

Dockerfile:
FROM python:3.6.8

RUN apt-get update && apt-get install -y --no-install-recommends libwebp-dev libjpeg-dev libpng-dev wget freetype2-demos libfreetype6-dev libfreetype6

RUN wget https://imagemagick.org/archive/ImageMagick.tar.gz --no-check-certificate && tar xvzf ImageMagick.tar.gz && rm ImageMagick.tar.gz && cd ImageMagick-7.1.0-51 && ./configure && make -j2 && make install && ldconfig /usr/local/lib && cd .. && rm -rf ImageMagick-7.1.0-51

**Comment by dlemstra** [comment, marked as answer]
Which version of freetype are you using. You should be using at least version 2.8.

**Reply by AbdulAzeezC** [reply]
Hi dlemstra,
sorry for writing here in the blog, i have similar issue while installing the magick lib,

i am using 7.1.0-57 latest version, but my freetype is 2.6 version, is there a way to upgrade or downgrade the version? or make it compatible with my current freetype version, by doing some change in "configure" file??

**Reply by dlemstra** [reply]
My advise would be to upgrade freetype. The version that you are using is from 2015 and probably contains bugs or security issues.
